cardazim/server.py

```diff
@@ -0,0 +1,97 @@
+import socket
+import argparse
+import sys
+from utils import unpack_message
+
+BACKLOG_SIZE = 1
+
+def read_all_data(socket: socket.socket) -> str:
+    """
+    Reads the message from the socket and then CLOSES it.
+    
+    :param socket: The socket from which the data is read.
+    :type socket.socket:
+    :returns: The data as an 'utf-8' string.
+    :rtype: str
+    """
+    from_client = b''
+    
+    with socket:
+        while True:
+            data = socket.recv(4096)
+            if not data:
+                break
+
+            from_client += data
+
+    return unpack_message(from_client)
+
+
+def init_server_socket(server_ip: str, server_port: int) -> None:
+    """
+    Creates and initializes a socket for the server.
+
+    :param server_ip: 
+    :type str:
+    :param server_port:
+    :type int:
+    """
+    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    server_socket.bind((server_ip, server_port))
+    server_socket.listen(BACKLOG_SIZE) # This is how many queued connections do we support.
+
+    return server_socket
+
+def listener_server(server_ip: str, server_port: int) -> None:
```

Comment:
this name implied that the function will return a server or something
maybe change the function name to something that implied what the function will do